frontend/templates/product-list/MetaTags.tsx

```diff
@@ -115,19 +114,14 @@ function useCanonicalUrl(
 ): string {
    const appContext = useAppContext();
    const pagination = usePagination();
-
-   const page = pagination.currentRefinement + 1;
-
-   const itemTypeHandle = itemType
-      ? `/${encodeURIComponent(stylizeDeviceItemType(itemType))}`
-      : '';
    const variant = useVariant();
+   const page = pagination.currentRefinement + 1;
 
-   return `${appContext.ifixitOrigin}${productListPath(
+   return `${appContext.ifixitOrigin}${productListPath({
       productList,
-      undefined,
-      productList.indexVariantsInsteadOfDevice ? variant : undefined
-   )}${itemTypeHandle}${page > 1 ? `?${PRODUCT_LIST_PAGE_PARAM}=${page}` : ''}`;
+      itemType: itemType ?? undefined,
+      variant: productList.indexVariantsInsteadOfDevice ? variant : undefined,
+   })}${page > 1 ? `?${PRODUCT_LIST_PAGE_PARAM}=${page}` : ''}`;
```

Comment:
I suspect that the increased complexity of `productListPath` function can (at least in part) be pushed into the product list model

> `itemType: itemType ?? undefined,`

`itemType` could be integrated into the product list model as an optional/nullable attribute. We already have `useItemTypeProductList` taking care of integrating item type override directly into the model (note that this happens to be a hook just because using Algolia hooks forces us to model it this way, otherwise that logic could be moved in the model).

> `variant: productList.indexVariantsInsteadOfDevice ? variant : undefined`

It looks like this ternary could be implemented directly inside `productListPath`.
Looking at the implementation though, it looks like that this kind of path that includes the variant is going to be used only in meta tags (btw, what's the reason for this behaviour?). If that's the case, we can create a specific function to compute the path and colocate it within this file, so that this logic is not made general purpose if it belongs only to a specific use case